vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -593,6 +586,24 @@ protected void onAttach(AttachEvent attachEvent) {
         initConnector();
     }
 
+    @Override
+    protected void onDetach(DetachEvent detachEvent) {
+        super.onDetach(detachEvent);
+        // When reloading a page using preserve on refresh, the notification
+        // should keep its opened state. To prevent it from auto-closing, delay
+        // the auto-closing logic to before client response, which is not called
+        // when reloading the page. This also prevents an exception when trying
+        // to remove an auto-added notification from its parent.
+        detachEvent.getUI().beforeClientResponse(this, executionContext -> {
```

Comment:
From my testing it seems this is enough to prevent the error reported in the linked issue, as the before client response is never called when reloading the page. I assume that's because the reload creates a new UI, and there never is a response for the old one that triggered the reload / detach.

This also ensures that the notification keeps its opened state, as the logic for auto-closing below is not executed.